state/accountsDB_test.go

```diff
@@ -2692,3 +2693,137 @@ func TestEmptyErrChanReturningHadContained(t *testing.T) {
 		})
 	})
 }
+
+func TestAccountsDb_Concurrent(t *testing.T) {
+	_, adb := getDefaultTrieAndAccountsDb()
+
+	numAccounts := 100000
+	accountsAddresses := generateAccounts(t, numAccounts, adb)
+	rootHash, _ := adb.Commit()
+
+	testAccountMethodsConcurrency(t, adb, accountsAddresses, rootHash)
+}
+
+func testAccountMethodsConcurrency(
+	t *testing.T,
+	adb state.AccountsAdapter,
+	addresses [][]byte,
+	rootHash []byte,
+) {
+	numOperations := 100
+	marshaller := &testscommon.MarshalizerMock{}
+	wg := sync.WaitGroup{}
+	wg.Add(numOperations)
+
+	numAccToLoad := 10
+	accounts := make([]vmcommon.AccountHandler, numAccToLoad)
+	for i := 0; i < numAccToLoad; i++ {
+		acc, err := adb.GetExistingAccount(addresses[i])
+		assert.Nil(t, err)
+		accounts[i] = acc
+	}
+	accountBytes, err := marshaller.Marshal(accounts[0])
+	assert.Nil(t, err)
+	counter := atomicFlag.Counter{}
```

Comment:
in this case we can safely remove counter and put wg.Done after the switch

Reply:
Removed.